### PR TITLE
658: Added button color, sizes, href, and base directions.

### PR DESCRIPTION
**Pull request by mescutia95**
I got most of what was needed done. Had issues getting the directional variation working as intended, so I left out my implementations for now. Changing it to left and right does not cause it to fall to the bottom anymore.

### CODE REVIEW

File: src/bossy-ui/components/dropdown/dropdown.component.ts [hunk 1]
```diff
@@ -10,12 +10,41 @@ export class BossyDropdownComponent implements OnInit {
   @Input() config: BossyDropdownConfig;
   showMenu = false;
   ariaExpanded = false;
+  btn_size: string = undefined;
```
Comment (tabersays) [inline src/bossy-ui/components/dropdown/dropdown.component.ts]: Would you please rename any snake cased variable to use lazy camel case. ie btn_size => btnSize

**Comment by glawson** [comment]
@mescutia95  - you have a unit test failing:

e 65.0.3325 (Linux 0.0.0): Executed 28 of 45 SUCCESS (0 secs / 0.602 secs)
Chrome 65.0.3325 (Linux 0.0.0) PopoverComponent should create FAILED
	[object ErrorEvent] thrown
Chrome 65.0.3325 (Linux 0.0.0): Executed 29 of 45 (1 FAILED) (0 secs / 0.631 secs)
Chrome 65.0.3325 (Linux 0.0.0) PopoverComponent should create FAILED
	[object ErrorEvent] thrown
e 65.0.3325 (Linux 0.0.0): Executed 30 of 45 (1 FAILED) (0 secs / 0.631 secs)
e 65.0.3325 (Linux 0.0.0): Executed 31 of 45 (1 FAILED) (0 secs / 0.631 secs)

**Comment by glawson** [comment]
@mescutia95 - yes, but that's kind of the point of the unit test.  Sometimes the changes you are working on might break other pieces of code, even outside what you touched.  But we also might have a case that the unit test was bad before your checking - I'm looking into that right now.

**Comment by glawson** [comment]
@tabersays - needs your approval

File: src/bossy-ui/components/dropdown/dropdown.component.ts [hunk 2]
```diff
@@ -10,12 +10,41 @@ export class BossyDropdownComponent implements OnInit {
   @Input() config: BossyDropdownConfig;
   showMenu = false;
   ariaExpanded = false;
+  btnSize: string = undefined;
```
Comment (tabersays) [inline src/bossy-ui/components/dropdown/dropdown.component.ts]: If these should be string wouldn't a default value of empty string make more sense?

**Comment by glawson** [comment]
resolves #658

**Pull request merged**